Remotion/Web/Core/UI/Controls/RendererBase.cs

```diff
@@ -144,13 +143,15 @@ protected IResourceManager GetResourceManager (Type localResourcesType, IResourc
       ArgumentUtility.CheckNotNull ("localResourcesType", localResourcesType);
       ArgumentUtility.CheckNotNull ("controlResourceManager", controlResourceManager);
 
-      var cachedResourceManager = _resourceManagerCache.GetOrAdd (
-          Tuple.Create (localResourcesType, controlResourceManager),
-          key => new Lazy<IResourceManager> (
-              () => ResourceManagerSet.Create (key.Item2, _globalizationService.GetResourceManager (key.Item1)),
-              LazyThreadSafetyMode.ExecutionAndPublication));
+      var result = _resourceManagerCache
+          .GetOrAdd (
+              localResourcesType,
+              _ => new ConditionalWeakTable<IResourceManager, Lazy<IResourceManager>>())
+          .GetValue (
```

Comment:
Please refactor to use double check:
first TryGetValue and if it's false, we do GetValue. Otherwise, we always create the closure during value access.

Reply:
Hmm there is another problem, if I'm not mistaken, delegate passed into the lazy also keeps a strong reference to the resource manager. The resource manager cannot be freed this way. I might need to change the type to `Lazy<WeakReference<IResourceManager>>` and adapt usages accordingly.